compiler/rustc_target/src/target_features.rs

```diff
@@ -590,9 +590,72 @@ const RISCV_FEATURES: &[(&str, StabilityUncomputed, ImpliedFeatures)] = &[
     // tidy-alphabetical-start
     ("a", STABLE, &["zaamo", "zalrsc"]),
     ("c", STABLE, &[]),
-    ("d", unstable(sym::riscv_target_feature), &["f"]),
-    ("e", unstable(sym::riscv_target_feature), &[]),
-    ("f", unstable(sym::riscv_target_feature), &[]),
+    (
+        "d",
+        Stability::Unstable {
+            nightly_feature: sym::riscv_target_feature,
+            allow_toggle: |target, enable| match &*target.llvm_abiname {
+                "ilp32d" | "lp64d" if !enable => {
```

Comment:
Btw do you know why the 32bit ABIs start with "ilp" but the 64bit ABIs start with "lp"?

Reply:
- ILP32: **I**nt, **L**ong, and **P**ointer are **32**-bit
- LP64: **L**ong and **P**ointer are **64**-bit (int is 32-bit)

(see also "Data models" section in https://en.cppreference.com/w/cpp/language/types)

Reply:
So it should really be I32LP64 then...

Reply:
Yes, but a convention at some point developed that
- we omit the "c8s16" prefix... types that are the smallest possible value
- we all quietly pretend the smallest possible value of int is 32 bits and not 16

names deeply fermented in tradition.